src/tests/ftest/util/apricot/apricot/test.py

```diff
@@ -140,30 +140,8 @@ def tearDown(self):
         if self.fault_file:
             os.remove(self.fault_file)
 
-
-class TestWithServers(TestWithoutServers):
-    """Run tests with DAOS servers and at least one client.
-
-    Optionally run DAOS clients on specified hosts.  By default run a single
-    DAOS client on the host executing the test.
-
-    :avocado: recursive
-    """
-
-    def __init__(self, *args, **kwargs):
-        """Initialize a TestWithServers object."""
-        super(TestWithServers, self).__init__(*args, **kwargs)
-
-        self.agent_sessions = None
-        self.setup_start_servers = True
-
-    def setUp(self):
-        """Set up each test case."""
-        super(TestWithServers, self).setUp()
-
-        self.server_group = self.params.get(
-            "name", "/server_config/", "daos_server")
-
+    def get_test_attributes(self):
```

Comment:
Yes, that is mainly the goal. Although after looking at the code for a bit, I think we can clean up how we get the attribute values from the yaml file. Since all I want to do essentially is grab the hosts values from the yaml file so that my tests can use the `self.hostlist_servers`, what about having a method to get the yaml value and a method to set the values of each attribute variable in `Test` class, similar to what is done in command_utils.py in `BasicParameter.get_yaml_value()` and in `ObjectWithParameters.get_attribute_names()`?  By doing this I can change the method name to something more specific: get_hosts_attributes() or 

    def get_yaml_value(self, name, path):
        return self.params.get(name, path)

    def set_attributes(self, path):
        for name, _ in self.__dict__.items():
            if self.get_yaml_value(name, path) != None:
                self.__dict__[name] = self.get_yaml_value(name, path)

Reply:
I'm on the fence.  I think it still feels misleading for a class not using `TestWithServer()s` to be using servers.  I'm not adamant about it, but just trying to avoid further complicating code if an existing mechanism works.

Looking at `TestWithServers()` a bit more, doesn't: https://github.com/daos-stack/daos/blob/5ff6429ed0397372b8e2796a6bb9f21c928a400e/src/tests/ftest/util/apricot/apricot/test.py#L217 (added in f3c7d7739) accomplish what you are looking for?  Perhaps you want to prevent the agents from being started also?  There is no reason a parallel `setup_start_agents` couldn't be easily) implemented as well.

Reply:
I see what you mean and I agree. I'll change the test so it inherits from TestWithServers.